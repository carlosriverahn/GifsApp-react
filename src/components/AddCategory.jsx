import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onChangeValue }) => {
	const [inputValue, setInputValue] = useState('');
	const onInputChage = ({ target }) => {
		// console.log(event);
		setInputValue(target.value);
	};

	const onSubmit = (event) => {
		event.preventDefault(event);
		if (inputValue.trim().length <= 1) {
			return;
		}

		onChangeValue(inputValue.trim());
		setInputValue('');
	};
	return (
		<form onSubmit={onSubmit} aria-label="form">
			<input
				type="text"
				placeholder="Search gifs"
				value={inputValue}
				onChange={(event) => onInputChage(event)}
			/>
		</form>
	);
};

AddCategory.propTypes = {
	onChangeValue: PropTypes.func.isRequired,
};
