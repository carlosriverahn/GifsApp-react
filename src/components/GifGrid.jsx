import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './gifItem';

export const GifGrid = ({ category }) => {
	const { images, isLoading } = useFetchGifs(category);
	
	return (
		<>
			<h3>Category</h3>
			{
				isLoading && ( <h2>Loading</h2> )
			}
			<div className="card-grid">
				{images.map((image) => (
					<GifItem key={image.id} image={image} />
				))}
			</div>
		</>
	);
};
