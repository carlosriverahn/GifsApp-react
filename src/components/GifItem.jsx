export const GifItem = ({ image }) => {
	return (
		<div className="card">
			<img src={image.urlImg} alt={image.title} />
			<p>{image.title}</p>
		</div>
	);
};
