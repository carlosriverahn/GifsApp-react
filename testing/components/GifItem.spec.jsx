import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/gifItem';

describe('Test in GifItem', () => {
	const title = 'Snapshot title';
	const urlImg = 'https://one-punch.com/saitama.jpg';

	test('should be match with the snapshot', () => {
		const { container } = render(<GifItem title={title} urlImg={urlImg} />);
		expect(container).toMatchSnapshot();
	});

	test('should be show the image with his url', () => {
		render(<GifItem title={title} urlImg={urlImg} />);
		// expect(screen.getByRole('img').src).toBe(urlImg);
		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(urlImg);
		expect(alt).toBe(title);
	});

	test('should be check if title exist', () => {
		render(<GifItem title={title} urlImg={urlImg} />);
		expect(screen.getByText(title)).toBeTruthy();
	});
});
