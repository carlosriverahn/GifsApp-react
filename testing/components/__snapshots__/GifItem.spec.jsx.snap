// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifItem should be match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Snapshot title"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Snapshot title
    </p>
  </div>
</div>
`;
